benchopt/base.py

```diff
@@ -517,3 +518,45 @@ def __reduce__(self):
             self._module_filename, module_hash, self._parameters, dataset,
             str(self._import_ctx._benchmark_dir)
         )
+
+    def default_split(self, *arrays):
+        i_train, i_test = next(self._cv)
+        res = ()
+        for x in arrays:
+            res = (*res, x[i_train], x[i_test])
+        return res
+
+    def custom_split(self, *arrays):
+        if not hasattr(self, "_cv"):
+            # this define _cv only once
+            self._cv = itertools.cycle(self.cv.split(*arrays))
+        i_train, i_test = next(self._cv)
+        return self.split(i_train, i_test)
+
+    def get_split(self, *arrays):
+        # return the split of the data according to the cv attribute
+
+        if not hasattr(self, "cv"):
+            raise ValueError(
+                "To use get_split, you need to define "
+                "a cv attribute in your objective class "
+                "It should be a sklearn.model_selection.BaseCrossValidator "
+                "object."
+            )
+
+        if not str(type(arrays[0])) == "<class 'numpy.ndarray'>":
```

Comment:
Even if the array is a `ndarray`, you might want a custom split, so I would not have this test.

Also, to properly test if you have an `ndarray`, you can use `isinstance(arrays[0], np.ndarray)`

Reply:
Also, typically, such split works with `pandas.DataFrame`  `torch.Tensor`, `sccipy.sparse.csc_matrix`, so it will be very hard to check if this is gonna work or not. I would just say, if no `split` is defined, just call the `default_split`.
If it does not work, raise an error saying that this is not compatible.

Reply:
Okay 👍  I look at this test also but I thought that it was less convenient because we need to import numpy

Reply:
Okay 👍 I looked at this test also but I thought that it was less convenient because we need to import numpy